Error while checking build/O2/fullCI for 6cf43dedd86a6f595309384c56f07646d825aec9 at 2023-06-27 21:55:
```
## sw/BUILD/o2checkcode-latest/log
--
========== List of errors found ==========
++ GRERR=0
++ grep -v clang-diagnostic-error error-log.txt
++ grep ' error:'
/sw/SOURCES/O2/11571-slc8_x86-64/0/Detectors/ITSMFT/ITS/tracking/GPU/cuda/TrackerTraitsGPU.cu:58:52: error: use nullptr [modernize-use-nullptr]
/sw/SOURCES/O2/11571-slc8_x86-64/0/Detectors/ITSMFT/ITS/tracking/GPU/hip/TrackerTraitsGPU.hip.cxx:58:51: error: use nullptr [modernize-use-nullptr]
++ [[ 0 == 0 ]]
++ exit 1
--
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/11571/6cf43dedd86a6f595309384c56f07646d825aec9/build_O2_fullCI/pretty.html).

--- a/Detectors/TRD/workflow/include/TRDWorkflow/NoiseCalibSpec.h
+++ b/Detectors/TRD/workflow/include/TRDWorkflow/NoiseCalibSpec.h
@@ -33,7 +33,7 @@ namespace trd
 class TRDNoiseCalibSpec : public o2::framework::Task
 {
  public:
-  TRDNoiseCalibSpec() = default;
+  TRDNoiseCalibSpec(bool dummy) : mIsDummy(dummy) {}
   void init(o2::framework::InitContext& ic) final
   {
     // Do we need to initialize something?
@@ -43,6 +43,10 @@ class TRDNoiseCalibSpec : public o2::framework::Task
   {
     auto digits = pc.inputs().get<gsl::span<Digit>>("trddigits");
     auto trigRecs = pc.inputs().get<gsl::span<TriggerRecord>>("trdtriggerrec");
+
+    if (mIsDummy) {
+      return;
+    }
 
     // Obtain rough time from the data header (done only once)
     if (mStartTime == 0) {
@@ -63,7 +67,7 @@ class TRDNoiseCalibSpec : public o2::framework::Task
         sendOutput(pc.outputs());
         mHaveSentOutput = true;
       } else {
-        if ((mNTFsProcessed % 50) == 0) {
+        if ((mNTFsProcessed % 200) == 0) {
           LOGP(important, "Not processing anymore. Seen {} TFs in total. Run can be stopped", mNTFsProcessed);
         }
       }
@@ -96,7 +100,9 @@ class TRDNoiseCalibSpec : public o2::framework::Task
     if (mHaveSentOutput) {
       LOGP(important, "Received EoS after sending calibration object. All OK");
     } else {
-      LOGP(alarm, "Received EoS before sending calibration object. Not enough digits received");
+      if (!mIsDummy) {
+        LOGP(alarm, "Received EoS before sending calibration object. Not enough digits received");
+      }
     }
   }
 
@@ -104,10 +110,11 @@ class TRDNoiseCalibSpec : public o2::framework::Task
   CalibratorNoise mCalibrator{};
   size_t mNTFsProcessed{0};
   bool mHaveSentOutput{false};
+  bool mIsDummy{false};
   uint64_t mStartTime{0};
 };
 
-o2::framework::DataProcessorSpec getTRDNoiseCalibSpec()
+o2::framework::DataProcessorSpec getTRDNoiseCalibSpec(bool dummy)
 {
   std::vector<InputSpec> inputs;
   inputs.emplace_back("trddigits", o2::header::gDataOriginTRD, "DIGITS", 0, Lifetime::Timeframe);
@@ -121,7 +128,7 @@ o2::framework::DataProcessorSpec getTRDNoiseCalibSpec()
     "trd-noise-calib",
     inputs,
     outputs,
-    AlgorithmSpec{adaptFromTask<TRDNoiseCalibSpec>()},
+    AlgorithmSpec{adaptFromTask<TRDNoiseCalibSpec>(dummy)},
     Options{}};
 }
 

--- a/Detectors/TRD/workflow/src/trd-calib-workflow.cxx
+++ b/Detectors/TRD/workflow/src/trd-calib-workflow.cxx
@@ -28,6 +28,7 @@ void customize(std::vector<o2::framework::ConfigParamSpec>& workflowOptions)
     {"vDriftAndExB", o2::framework::VariantType::Bool, false, {"enable vDrift and ExB calibration"}},
     {"noise", o2::framework::VariantType::Bool, false, {"enable noise and pad status calibration"}},
     {"gain", o2::framework::VariantType::Bool, false, {"enable gain calibration"}},
+    {"calib-dds-collection-index", VariantType::Int, -1, {"allow only single collection to produce calibration objects (use -1 for no limit)"}},
     {"configKeyValues", VariantType::String, "", {"Semicolon separated key=value strings"}}};
 
   std::swap(workflowOptions, options);
@@ -55,7 +56,21 @@ WorkflowSpec defineDataProcessing(ConfigContext const& configcontext)
     if (enableRootInp) {
       specs.emplace_back(o2::trd::getTRDDigitReaderSpec(false));
     }
-    specs.emplace_back(o2::trd::getTRDNoiseCalibSpec());
+    int ddsCollectionIdx = configcontext.options().get<int>("calib-dds-collection-index");
+    bool noiseCalibIsDummy = true;
+    if (ddsCollectionIdx != -1) {
+      char* colIdx = getenv("DDS_COLLECTION_INDEX");
+      int myIdx = colIdx ? atoi(colIdx) : -1;
+      if (myIdx == ddsCollectionIdx) {
+        LOG(info) << "TRD noise calib is enabled for this collection, my index " << myIdx;
+        noiseCalibIsDummy = false;
+      } else {
+        LOG(info) << "TRD noise calib is disabled for this collection, my index " << myIdx;
+      }
+    } else {
+      noiseCalibIsDummy = false;
+    }
+    specs.emplace_back(o2::trd::getTRDNoiseCalibSpec(noiseCalibIsDummy));
   }
 
   if (configcontext.options().get<bool>("gain")) {

